packages/castmill/lib/castmill/resources.ex

```diff
@@ -579,9 +581,80 @@ defmodule Castmill.Resources do
   it will trigger a "delete" webhook event if such a webhook is configured.
   """
   def delete_media(%Media{} = media) do
-    # TODO: Call relevant webhooks so that the integration has a chance to
-    # clean up the media from the storage system.
-    Repo.delete(media)
+    # TODO: check if the media is used somewhere, if so, it cannot be
+    # deleted until the references are removed. (Maybe we can force
+    # delete the media and remove all itsreferences as well in this call)
+
+    # Get all the files belonging to the media
+    media = get_media(media.id)
+
+    # Iterate through all the files and delete them
+    Enum.each(media.files, fn {_context, file} ->
```

Comment:
This seems to delete the files from S3 before the referring database entries are deleted. Shouldn't we do this the other way around? Better to leave orphan medias than having dangling file references in the database if the server dies in between.

Reply:
I thought that S3 is more expensive than a dangling file row in postgresql and that it is easier to clean up. A dangling S3 file will be impossible to map to an owner.